client/src/features/product.js

```diff
@@ -42,6 +46,42 @@ export const setProductsByFavoredUserId = createAsyncThunk('products/setByFavore
   }
 })
 
+export const setProductWithNewAnswer = createAsyncThunk('products/setAnswer', async ({ token, params }) => {
```

Comment:
Function name "set..." is used to fetch data from DB and put them in redux. Not suitable for update.